src/frontend/App.tsx

```diff
@@ -56,7 +57,9 @@ const App = () => {
               <ActiveProjectProvider>
                 <PhotoPromiseProvider>
                   <SecurityProvider>
-                    <AppNavigator permissionAsked={permissionsAsked} />
+                    <React.Suspense fallback={<Loading />}>
+                      <AppNavigator permissionAsked={permissionsAsked} />
+                    </React.Suspense>
```

Comment:
not familiar with suspense enough to really understand why this is needed - but it definitely sticks out as something i'd have additional questions about. guess my main question would be: what is this suspending on, and is this the preferred way of doing it vs some conditional rendering in one of the wrapping providers?

Reply:
Previously this `<React.Suspense />` was inside the App Navigator. Any useSuspenseQuery(), triggers a loading state, and then the fallback component in the nearest suspense component up the tree is shown to the user until loading is done.

So the reason I had to move it out of the App Navigator, is that component needs access to all the presets (in order to check if the saved observations has an exisiting preset, which uses the `presetMatcher` function). The presets are retrieved through a useSuspenseQuery. The suspense has to be a parent component to any suspenseQuery.

The suspense API is now a stable API, and the [caveats of use](https://react.dev/reference/react/Suspense#caveats) don't apply to anything we do. Overall it is also a much cleaner code as opposed to conditional rendering. 

I think i made a comment the other day that suspense is not the preferred API, but i was confusing it with `ErrorBoundary`...

Reply:
thanks for the explanation!